pilot/pkg/config/coredatamodel/controller.go

```diff
@@ -176,18 +179,26 @@ func (c *Controller) Apply(change *sink.Change) error {
 	}
 
 	var prevStore map[string]map[string]*model.Config
-
 	c.configStoreMu.Lock()
 	prevStore = c.configStore[descriptor.Type]
 	c.configStore[descriptor.Type] = innerStore
```

Comment:
Is this still correct if we're doing incremental EDS? I suspect that this code was assuming that we're getting *all* of the resources in the snapshot.  With incremental, that's no longer the case.

Reply:
I will update as per our discussion.